### PR TITLE
Fix android window focus

### LINKED ISSUE #2385: Android: Window focus does not update during app life cycles
Android Game Window `isActive` is never updated when the app moves between its relevant view cycles, and is stuck as `false`

Fixed in PR: #2384

### PR DESCRIPTION
**Pull request by SilkyPantsDan**
Ensures that the window focus is updated when app moves into foreground and background. This should also trigger the Pause menu in the main osu! game when required. 

Resolves #2385

### CODE REVIEW

**Comment by SilkyPantsDan** [comment]
Resolves #2385

File: osu.Framework.Android/AndroidGameWindow.cs [hunk 1]
```diff
@@ -16,8 +16,18 @@ public override IGraphicsContext Context
 
         internal static AndroidGameView View;
 
+        private bool activityInForeground = true;
+        public bool ActivityInForeground
```
Comment (UselessToucan) [inline osu.Framework.Android/AndroidGameWindow.cs]: Why did you add new property instead of using `Focused` getter/setter?

Comment (SilkyPantsDan) [reply osu.Framework.Android/AndroidGameWindow.cs]: Still getting my head around all the bindings etc, but the `Focused` property appears to only be a get/read only in the base class.

File: osu.Framework/Platform/GameWindow.cs [hunk 2]
```diff
@@ -84,7 +84,7 @@ protected GameWindow([NotNull] IGameWindow implementation)
             MouseEnter += (sender, args) => CursorInWindow = true;
             MouseLeave += (sender, args) => CursorInWindow = false;
 
-            FocusedChanged += (o, e) => isActive.Value = Focused;
+            FocusedChanged += OnFocusedChanged;
```
Comment (peppy) [inline osu.Framework/Platform/GameWindow.cs]: This change doesn't seem to be doing anything. Any specific reason for it?

Comment (SilkyPantsDan) [reply osu.Framework/Platform/GameWindow.cs]: Only so the `OnFocusedChanged` can be triggered when the background updates as well - but the more I'm learning about the framework etc, the more I think this could probably all be handled with the Bindable stuff too - just haven't had a chance to get back to this yet

Comment (peppy) [reply osu.Framework/Platform/GameWindow.cs]: Aha, I see.

Yeah, these changes are hard to follow, possibly due to the naming being different in different classes.

**Comment by SilkyPantsDan** [comment]
Actually - going to close this and look at it from a different point, with the knowledge I have now too. It seems that the `Focused` property is actually coming from the OpenTK/osuTK implementation so it might be better to place these down there for the moment, while also looking at the background issue.
